src/ol/interaction/drawinteraction.js

```diff
@@ -674,7 +678,7 @@ ol.interaction.Draw.prototype.abortDrawing_ = function() {
     this.sketchFeature_ = null;
     this.sketchPoint_ = null;
     this.sketchLine_ = null;
-    this.overlay_.getFeatures().clear();
+    this.overlay_.getSource().clear();
```

Comment:
I can try that. I'll also add a test to make sure that a features collection will stay in sync.